openwisp_monitoring/device/tests/test_admin.py

```diff
@@ -345,6 +345,195 @@ def test_wifisession_inline(self):
             self.assertNotContains(response, '<h2>WiFi Sessions</h2>')
             self.assertNotContains(response, 'monitoring-wifisession-changelist-url')
 
+    def test_check_alertsetting_inline(self):
+        test_user = self._create_user(
+            username='test', email='test@inline.com', is_staff=True
+        )
+        self._create_org_user(is_admin=True, user=test_user)
+        device = self._create_device()
+        ping_check = Check(
+            check_type=CHECK_CLASSES[0][0], content_object=device, params={}
+        )
+        ping_check.full_clean()
+        ping_check.save()
+        url = reverse('admin:config_device_change', args=[device.pk])
+        metric = self._create_general_metric(
+            name='', content_object=device, configuration='ping'
+        )
+        self._create_alert_settings(metric=metric)
+        self.client.force_login(test_user)
+
+        def _assert_check_inline_in_response(response):
+            self.assertContains(response, '<h2>Checks</h2>', html=True)
+            self.assertContains(response, 'check-content_type-object_id-0-is_active')
+            self.assertContains(response, 'check-content_type-object_id-0-check_type')
+            self.assertContains(response, 'check-content_type-object_id-0-DELETE')
+
+        def _assert_alertsettings_inline_in_response(response):
+            self.assertContains(response, '<h2>Alert Settings</h2>', html=True)
+            self.assertContains(
+                response,
+                '<img src="/static/admin/img/icon-yes.svg" alt="True">',
+                html=True,
+            )
+            self.assertContains(
+                response, 'metric-content_type-object_id-0-configuration'
+            )
+            self.assertContains(
+                response,
+                'metric-content_type-object_id-0-alertsettings-0-is_active',
+            )
+            self.assertContains(
+                response, '<option value="&lt;" selected>less than</option>'
+            )
+            self.assertContains(
+                response,
+                'metric-content_type-object_id-0-alertsettings-0-custom_threshold" value="1"',
+            )
+            self.assertContains(
+                response,
+                'metric-content_type-object_id-0-alertsettings-0-custom_tolerance" value="0"',
+            )
+            self.assertContains(
+                response,
+                'metric-content_type-object_id-0-alertsettings-0-DELETE',
+            )
+
+        with self.subTest('Test check & alert settings with model permissions'):
+            test_user_permissions = test_user.user_permissions.all().values_list(
+                'codename', flat=True
+            )
+            self.assertEqual(test_user_permissions.count(), 0)
```

Comment:
I would re-write this as following 

```suggestion
            self.assertEqual(test_user.user_permissions.count(), 0)
```

You are not using the content of the returned list, hence there's no point in requesting that data from the database. Please update other occurrences as well.